### PR TITLE
support r/w floating(single) value type (CON-1077)

### PR DESCRIPTION
**Pull request by YOGYUI**
@dhrishi I recently lost my remote and local branch while squashing PR commits because of my mistake, so reopened PR with new branch. 
I deeply sorry for inconvenience.

[Related issue]
- https://github.com/espressif/esp-matter/issues/843

[Major changes]
- support read/write floating (single) attribute value type
- make `ConcentrationMeasurement` clusters follow Matter specifications (2.10.6)
    - change attribute value type (`MeasuredValue`, `MinMeasuredValue`, `MaxMeasuredValue`, `PeakMeasuredValue`, `AverageMeasuredValue`, `Uncertainty`)
    - remove creating `Uncertainty` attribute phrase (optional attribute)
    - add creating `MeasurementMedium` attribute phrase (mandatory attribute)
    - add `AverageMeasurement` feature config struct

[Minor changes]
- modify indentation (tab to 4 spaces)

### CODE REVIEW

**Comment by dhrishi** [comment]
Sure. No worries. Thanks @YOGYUI

File: components/esp_matter/esp_matter_attribute.cpp [hunk 1]
```diff
@@ -851,49 +851,49 @@ namespace attribute {
 attribute_t *create_phy_rate(cluster_t *cluster, nullable<uint8_t> value)
 {
     return esp_matter::attribute::create(cluster, EthernetNetworkDiagnostics::Attributes::PHYRate::Id,
-					 ATTRIBUTE_FLAG_NULLABLE, esp_matter_nullable_enum8(value));
+                                         ATTRIBUTE_FLAG_NULLABLE, esp_matter_nullable_enum8(value));
```
Comment (dhrishi) [inline components/esp_matter/esp_matter_attribute.cpp:854]: Please remove all such changes from your diff

**Review by wqx6 — APPROVED** [review]
LGTM

**Review by jadhavrohit924 — APPROVED** [review]

**Comment by lboue** [comment]
Can you merge this PR is it's OK?

**Comment by YOGYUI** [comment]
@dhrishi Any progress?

**Comment by dhrishi** [comment]
@YOGYUI @lboue Sorry for the delay. We will soon have this in the internal queue and then sync to Github.

**Pull request merged**